.github/workflows/shiny-deploy.yaml

```diff
@@ -0,0 +1,48 @@
+# Workflow derived from https://github.com/r-lib/actions/tree/v2/examples
+# see https://www.tidyverse.org/blog/2022/06/actions-2-0-0/ for info on setup-r-dependencies@v2
+on:
+  push:
+    branches: [production]
+  workflow_dispatch:
+
+name: shiny-deploy
+
+jobs:
+  shiny-deploy:
+    runs-on: ubuntu-latest
+    env:
+      GITHUB_PAT: ${{ secrets.GITHUB_TOKEN }}
+    steps:
+      - uses: actions/checkout@v2
+
+      - uses: r-lib/actions/setup-pandoc@v2
+
+      - uses: r-lib/actions/setup-r@v2
+        with:
+          use-public-rspm: true
+
+      - uses: r-lib/actions/setup-r-dependencies@v2
+        with:
+          cache-version: 2
+          extra-packages: |
+            any::rsconnect
+
+      - name: Authorize and deploy app
+        env:
+          # Provide your app name, account name, and server to be deployed below
+          APPNAME: ${{ secrets.RSCONNECT_APPNAME }}
+          ACCOUNT: ${{ secrets.RSCONNECT_USER }}
+          SERVER: ${{ secrets.RSCONNECT_SERVER }}
+        run: |
```

Comment:
Could this be a simple command to run a script rather than the full script within yaml? I think it will be easier to maintain an R code file rather than R code embedded within yaml.

Reply:
It could be, but I was wondering if you might want to drop or change the nature of the email generation support to something that removes this 3rd party dependency. The mailR package requires Java.

Hopefully this code doesn't require any maintenance other than simplification.

Reply:
I would still prefer to have this setup the environment variables from the github secrets and then call `scripts/deploy.R`. Even though it's only a few lines of code here, it may become more complicated, code highlighting works, we could add tests, etc. 

Separately, it looks like this code also doesn't copy the full appfiles. https://github.com/StevenLShafer/stanpumpR/blob/00e37be3d48f5ce6f665948b8718e69e2a5f8bc3/scripts/deploy.R#L28-L30

I was able to setup the environment variables from travis, so once that change is made I'm ready to merge and test it out.

With regards to changing the email generation, good question, but let's pause on that for this PR which will migrate the deployment from travis to github actions.

Reply:
You don't need to specify the appFiles unless you want to exclude certain files from upload.

From the deployApp {rsconnect} documentation:

appFiles:
"The files and directories to bundle and deploy (only if upload = TRUE). Can be NULL, in which case all the files in the directory containing the application are bundled, with the exception of any listed in an .rscignore file. Takes precedence over appFileManifest if both are supplied."